enhancements/ingress/set-replace-append-delete-deny-headers.md

```diff
@@ -0,0 +1,877 @@
+  Answer: Yes, you would be able to set a minimal set of dynamic header values. We had three options for this EP.
+    1. Allow dynamic header values with all sample fetchers and converters.
+    2. Allow a minimal set of dynamic values that we can confidently validate, focusing on the things that people have asked for on the RFE or [EP](https://github.com/openshift/enhancements/pull/1296#issuecomment-1396103783).
+    3. Prohibit dynamic values entirely.
+    However, as few customers asked for it, we went ahead with option number 2.
+    For setting dynamic header values for http response. Allowed sample fetchers are res.hdr, ssl_c_der. Allowed converters are lower, base64.
+    For setting dynamic header values for http request. Allowed sample fetchers are req.hdr, ssl_c_der. Allowed converters are lower, base64.
+    Example: One might be able to set value of `x-forwarded-client-cert` to `%{+Q}[ssl_c_der,base64]`.
+
+- Certain headers like HSTS are not allowed to be set using this API as there is already a route annotation available for this
+  Would it not be better to provide a concrete API and deprecate the annotation API?
+  Answer: The strict-transport-security header (AKA HSTS) has wider ranging implications than the typical HTTP header. We considered allowing HSTS to be configured
+    on the IngressController in the past; see the global-options-enable-hsts EP, which was never implemented. However, HSTS isn't neatly scoped: Route A's HSTS
+    policy can affect Route B, even if the routes are in different namespaces or belong to different IngressControllers, if Route B's host name is a subdomain of
+    Route A's. Additionally, specifying HSTS on an IngressController can cause a route manifest that behaves as expected on one cluster to behave surprisingly and
+    problematically on another cluster. For these reasons, we decided not to implement the global-options-enable-hsts EP and instead wrote and implemented the
+    global-admission-hsts EP, which defines a cluster-scoped API, using which the cluster admin can specify HSTS parameters that may or must be specified on routes
+    for a given domain: https://github.com/openshift/api/blob/6ba31fa438f20d6c822ef47ad3d771309cb5216b/config/v1/types_ingress.go#L64-L87. Based on this experience,
+    we wouldn't want to allow configuring HSTS on the IngressController using this new API.
+
+### Test Plan
+
+The controller that manages the IngressController Deployment and related
+resources has unit test coverage; for this enhancement, the unit tests are
+expanded to cover the additional functionality.
+
+The operator has end-to-end tests; for this enhancement, a test is added that
+(1) creates an IngressController that specifies certain response
+headers should be set or deleted, (2) sends an HTTP request to one of the standard
+routes (such as the console route), and (3) verifies that the expected headers
+are seen in the `curl -vk <route host>`.
```

Comment:
This all looks reasonable.  You can also mention that integration tests in openshift/api, the unit tests in library-go, and the unit tests for the base64-encoding and quoting logic.